### ISSUE TITLE
Error if calling a string/template/number literal

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#611]
```js
let randomVar = 5();
```
5 is called as if it was a function. We should report an error for trivial cases like this.

In the future, we might want to provide a nice diagnostic suggesting inserting a semicolon in cases like the following (suggested by elliot416):
```js
let randomVar = 5 // <-- no semicolon
 
(function() {
 
});
```

**Comment by Ishvina** [comment]
Hi! I am in an intro to swe class that is requiring me to work on my first open-source project. I was wondering if this issue is still open and if so could I get help with my first contribution?

**Comment by strager** [comment]
Yes, this task is still open.

I can definitely help you with this task if you have problems. I can answer questions here or weekdays on my Twitch stream (https://twitch.tv/strager). (However, I'm on a vacation right now, so my availability is spotty until January 26.)

cc @singalhimanshu who can help if I'm not around.

**Comment by Ishvina** [comment]
Ok perfect thank you so much!

**Comment by singalhimanshu** [comment]
@Ishvina let me know if you are facing any issues.

**Comment by Ishvina** [comment]
@singalhimanshu Hi! Thank you. Can you tell me where to begin and where I should find these errors?

**Comment by singalhimanshu** [comment]
> @singalhimanshu Hi! Thank you. Can you tell me where to begin and where I should find these errors?

Have you done the project setup? Are you able to run quick-lint-js on js files? If not follow the instructions here - https://github.com/quick-lint/quick-lint-js/blob/master/docs/BUILDING.md

**Comment by strager** [comment]
> Can you tell me where to begin and where I should find these errors?

I usually start by reproducing the problem. In this case, the problem is that quick-lint-js reports *no* error, but Node.js does report an error (when the code runs). I'd reproduce by creating a .js file and running quick-lint-js and Node.js on the .js file. (You need to build quick-lint-js from source by following the instructions @singalhimanshu linked.)

Then, I'd write a test case. I'd probably write a test in `test/test-parse-expression.cpp`. In order to write the test, we need a new error type, which we'd add in `src/quick-lint-js/error.h`.

**Comment by mketer1** [comment]
Hi! I am taking an intro class that requires me to work on my first open-source project. I noticed that this issue was assigned two months ago but has not been resolved. If this issue is still open, could I be assigned it?

**Comment by strager** [comment]
@Ishvina Are you still working on this task?

**Comment by mketer1** [comment]
If @Ishvina does not respond, would I be able to receive assignment for this task. Also, because I'm still new to open source contribution, would it be possible to get guidance/assistance if need be?

**Comment by strager** [comment]
@mketer1 Do you still want to work on this task?

**Comment by fk00750** [comment]
Hey @strager can I work on this task?

**Comment by strager** [comment]
@fk00750 You may.

**Comment by strager** [comment]
@fk00750 Are you still working on this task? Do you need any help?

**Comment by catmar22** [comment]
Hello! Has this been resolved yet? If not I'm also interested in trying to solve this

**Comment by strager** [comment]
@catmar22 I think @fk00750 abandoned the task. You can take it over.

**Comment by adrperez5** [comment]
Hi @strager, has this issue been resolved? I would like to give it a go.

**Comment by strager** [comment]
@adrperez5 It looks like there were a lot of people who accepted the task then disappeared. Maybe this task is haunted... Are you sure you want to take it on?

**Comment by adrperez5** [comment]
I will try my best!
